crates/matrix-sdk-base/src/client.rs

```diff
@@ -556,66 +556,84 @@ impl BaseClient {
+                changes.add_room(room_info);
+            }
 
-        let now = Instant::now();
+            self.handle_invited_state(&new_info.invite_state.events, &mut room_info, changes);
 
-        #[cfg(feature = "e2e-encryption")]
-        let to_device = {
-            if let Some(o) = self.olm_machine() {
-                // Let the crypto machine handle the sync response, this
-                // decrypts to-device events, but leaves room events alone.
-                // This makes sure that we have the decryption keys for the room
-                // events at hand.
-                o.receive_sync_changes(
-                    to_device,
-                    &device_lists,
-                    &device_one_time_keys_count,
-                    device_unused_fallback_key_types.as_deref(),
-                )
-                .await?
-            } else {
-                to_device
-            }
-        };
+            changes.add_stripped_room(room_info);
 
-        let mut changes = StateChanges::new(next_batch.clone());
-        let mut ambiguity_cache = AmbiguityCache::new(self.store.inner.clone());
+            new_rooms.invite.insert(room_id, new_info);
+        }
+        Ok(())
+    }
 
-        self.handle_account_data(&account_data.events, &mut changes).await;
+    /// Processes everything from the `/sync` `leave` timeline.
+    async fn process_left_rooms(
+        &self,
+        left_rooms: BTreeMap<OwnedRoomId, api::sync::sync_events::v3::LeftRoom>,
+        changes: &mut StateChanges,
+        ambiguity_cache: &mut AmbiguityCache,
+        push_rules: &Ruleset,
+        new_rooms: &mut Rooms,
+    ) -> Result<()> {
+        for (room_id, new_info) in left_rooms {
+            let room = self.store.get_or_create_room(&room_id, RoomType::Left).await;
+            let mut room_info = room.clone_info();
+            room_info.mark_as_left();
 
-        let push_rules = self.get_push_rules(&changes).await?;
+            let mut user_ids = self
+                .handle_state(&new_info.state.events, &mut room_info, changes, ambiguity_cache)
+                .await?;
 
-        let mut new_rooms = Rooms::default();
+            let timeline = self
+                .handle_timeline(
+                    &room,
+                    new_info.timeline,
+                    push_rules,
+                    &mut user_ids,
+                    &mut room_info,
+                    changes,
+                    ambiguity_cache,
+                )
+                .await?;
+
+            self.handle_room_account_data(&room_id, &new_info.account_data.events, changes).await;
+
+            changes.add_room(room_info);
+            new_rooms
+                .leave
+                .insert(room_id, LeftRoom::new(timeline, new_info.state, new_info.account_data));
+        }
+        Ok(())
+    }
 
-        for (room_id, new_info) in rooms.join {
+    /// Processes everything from the `/sync` `join` timeline.
+    #[allow(unused_variables)]
```

Comment:
Doing this for the whole function seems wrong. I guess this is because of code behind `#[cfg]`s? Can you just put this attribute on the relevant `let` bindings then?

Reply:
Actually I think this is a left over of an earlier iteration. I will check and remove it.